asu/build.py

```diff
@@ -282,6 +262,27 @@ def download_file(filename: str, dest: str = None):
 
     log.debug("Created store path: %s", req["store_path"] / bin_dir)
 
+    config_path = cache / subtarget / ".config"
+    config = config_path.read_text()
+
+    # enables all filesystem if none is specified
+    for filesystem in ["squashfs", "ext4fs", "ubifs", "jffs2"]:
+        # this implementation uses `startswith` since a running device thinks
+        # it's running `ext4` or `ubi` while it's really called `ext4fs`
```

Comment:
This is true for `ext4` which becomes `ext4fs`. However, `ubifs` is just called `ubifs` always (the backing device for UBIFS is always a UBI *volume* which lives on a UBI device which typically lives on a MTD partition of a NAND MTD device). So using `startswith` is of course ok, just the comment needs to be more clear about the reason being `ext4fs` / `ext4` particularly and *not* some generic rule for which `ubifs` / `ubi` would be another example (because as stated above, it is not. The filesystem is also called `ubifs` by the Linux kernel.)

Sidenode: The `-ubi` in the image name of some devices does **not** indicate that they are using UBIFS as rootfs. Production systems using UBI to manage NAND storage will just squashfs, just like you'd do on x86.